entity-service/src/integrationTest/java/org/hypertrace/entity/service/service/EntityQueryServiceTest.java

```diff
@@ -734,4 +788,10 @@ private static Datastore getDatastore() {
 
     return attributesMap;
   }
+
+  private <T> void assertListEquals(List<T> expected, List<T> actual) {
```

Comment:
why can't this just use assertEquals?

Reply:
I tried that. It wasn't working with the basic `assertEquals`, though it should have

Didn't spend much time on why it wasn't working

Reply:
IMO, worth figuring out - something's off here. This check for example doesn't assert that their order matches, which is part of list equality.